redis/src/cmd.rs

```diff
@@ -601,6 +630,63 @@ impl Cmd {
     pub fn is_no_response(&self) -> bool {
         self.no_response
     }
+
+    /// It marks command as triable for client side caching.
+    /// Useful when `CacheMode` is set to `CacheMode::OptIn`
+    #[cfg(feature = "cache-aio")]
+    pub fn cache(&mut self) -> &mut Cmd {
+        self.cache = Some(CommandCacheOptions {
+            client_side_ttl: None,
+        });
+        self
+    }
+    /// It marks command as triable for client side caching with custom Client Side TTL for cache.
+    /// Useful when `CacheMode` is set to `CacheMode::OptIn`
+    #[cfg(feature = "cache-aio")]
+    pub fn cache_with_client_side_ttl(&mut self, client_side_ttl: Duration) -> &mut Cmd {
+        self.cache = Some(CommandCacheOptions {
+            client_side_ttl: Some(client_side_ttl),
+        });
+        self
+    }
+    #[cfg(feature = "cache-aio")]
+    #[inline]
+    pub(crate) fn compute_cache_information(&self) -> Option<CommandCacheInformationByRef> {
+        let mut is_mget = false;
+        let x: &[u8] = &[];
+        if self.args.len() >= 2 {
+            if let Some(command_name) = self.arg_idx(0) {
+                let cache_key = match command_name {
+                    b"GET" | b"HEXISTS" | b"HGET" | b"HMGET" | b"HGETALL" | b"PTTL" => {
+                        Some(self.arg_idx(1).unwrap())
+                    }
+                    b"MGET" => {
+                        is_mget = true;
+                        Some(x)
+                    }
+                    _ => return None,
```

Comment:
why other commands can't be cached?

Reply:
Only read commands *should* be cached, I can prepare extensive list but this is kinda MVP.

Reply:
Hmm NOLOOP option was never on my plans, I will create an issue about it.
there was an early issue about it when it was mostly draft (https://github.com/redis-rs/redis-rs/issues/826)

Reply:
https://github.com/redis-rs/redis-rs/issues/1316